src/detect-http2.c

```diff
@@ -806,6 +806,8 @@ static int DetectEngineInspectHttp2HeaderName(
 
 static int DetectHTTP2headerSetup(DetectEngineCtx *de_ctx, Signature *s, const char *arg)
 {
+    SCLogWarning(SC_WARN_DEPRECATED, "Keyword http2.header is removed in Suricata 7, and replaced "
+                                     "by http.request_header and http.response_header");
```

Comment:
This does seem a little *loud*, but it might not matter.

- Has `http2.header` *only* existed for Suricata 6?
- I don't see any public rule sources using this keyword.

If both the above are true, then I am OK with being this loud. Otherwise probably a warning.

Reply:
> Has http2.header only existed for Suricata 6?

Yes

> I don't see any public rule sources using this keyword.

Did you just run a grep ?

Reply:
Pretty much.. I index all public rules from our index in a database, and I checked that.